src/app/query/bitcoin/address/transactions-by-address.hooks.ts

```diff
@@ -33,34 +35,40 @@ export function useBitcoinPendingTransactionsInputs(address: string) {
   });
 }
 
-function useFilterAddressPendingTxsOutputs(address: string) {
-  return useCallback(
-    (pendingTxs: BitcoinTransaction[]) => {
-      return pendingTxs.flatMap(tx => {
-        const inputsFromAddress = tx.vin.filter(
-          input => input.prevout.scriptpubkey_address === address
-        );
-        // Output is possibly change, so we only subtract the value if the address
-        // is funding the tx and sending utxos to a different address
-        return tx.vout.filter(
-          output => inputsFromAddress.length && output.scriptpubkey_address !== address
-        );
-      });
-    },
-    [address]
-  );
+export function calculatePendingTxsValue(pendingTxs: BitcoinTransaction[], address: string) {
+  // sum all inputs
+  const sumInputs = sumNumbers(pendingTxs.flatMap(tx => tx.vin.map(input => input.prevout.value)));
+
+  // get all outputs that are sent back to the address
+  const filteredOutputs = pendingTxs
+    .flatMap(tx => tx.vout.map(output => output))
+    .filter(v => v.scriptpubkey_address === address)
```

Comment:
How do you know this output isn't an address receiving btc? I created `useFilterAddressPendingTxsOutputs` bc I think you have to check first if the address is actually funding the tx as the `sender` to know it is change? Basically, how do you know this is being `sent back` rather than just `sent to` this address?

Reply:
I added `filterMissingUtxosPendingTxs` to check that we subtract from balance existing utxos. otherwise we would subtract same tx twice (if pending tx still exists, and there are no spent utxos in list). 
we get info about utxos and pending txs from different apis, that's why there might be such collision.

I think `filterMissingUtxosPendingTxs` covers case you described?